python/lib/metrics/dmod/metrics/categorical.py

```diff
@@ -0,0 +1,1036 @@
+"""
+Defines methods and structures that facilitate the evaluation of categorical metrics,
+such as Probability of Detection and False alarm ratio
+
+Attributes:
+    PANDAS_DATA: A type indicating either a pandas DataFrame or a pandas Series
+    NUMBER: A type indicating either a floating point or an integer
+"""
+import typing
+import math
+
+import pandas
+import numpy
+
+from dmod.metrics.threshold import Threshold
+
+PANDAS_DATA = typing.Union[pandas.DataFrame, pandas.Series]
+
+NUMBER = typing.Union[float, int]
```

Comment:
Oddly enough a bunch of functions in `math` don't acknowledge `numbers.Number` (though they are always supported). As a result, it throws a fit when linting. Mind if I try `typing.SupportsFloat`? That's what the `math` functions apparently expect.

Reply:
Ah. And... `typing.SupportsFloat` doesn't define operators so the linter _really_ doesn't like that. The union does really well because all the `int`, `float`, and all the similar `numpy` dtypes fit under that umbrella as well.

Reply:
Very weird, but that certainly counts as a reason to go this route. I'll resolve this item.